src/renderer.js

```diff
@@ -41,6 +41,58 @@ document.addEventListener("DOMContentLoaded", () => {
       ipcRenderer.send("new-window");
     });
 
+    nav.addEventListener("toggle-bookmark", async () => {
+      const urlInput = nav.querySelector("#url");
+      if (!urlInput || !urlInput.value) {
+        console.error("URL input is empty, cannot toggle bookmark.");
+        return;
+      }
+      const url = urlInput.value.trim();
+      
+      const bookmarks = await ipcRenderer.invoke("get-bookmarks");
+      const existingBookmark = bookmarks.find((b) => b.url === url);
+
+      if (existingBookmark) {
+        ipcRenderer.invoke("delete-bookmark", { url });
+      } else {
+        const title = pageTitle.innerText
+          .replace(" - Peersky Browser", "")
+          .trim();
+        let favicon = "";
+        try {
+          const iconLink = document.querySelector(
+            'link[rel="icon"], link[rel="shortcut icon"]'
+          );
+          if (iconLink) {
+            favicon = new URL(iconLink.href, url.origin).href;
+          }
+
+          if (!favicon) {
+        
+            favicon = `https://www.google.com/s2/favicons?domain=${url.hostname}`;
+          }
+        } catch (e) {
+          console.error("Error fetching favicon:", e);
+          favicon = `https://www.google.com/s2/favicons?domain=${url.hostname}`;
+        }
+        ipcRenderer.send("add-bookmark", { url, title, favicon });
```

Comment:
You're bookmarking whatever is currently in the URL bar (`#url` input in the `#navbox`), which can contain internal URLs like `peersky://home`.

<img width="743" height="167" alt="Screenshot 2025-07-12 at 2 28 22 PM" src="https://github.com/user-attachments/assets/b4f37a19-372c-4a63-9928-92d418e6e7bc" />

Behaviour:
- You open your browser, it loads peersky://home.
- You hit the bookmark button.
- The code grabs the value from the URL input: peersky://home.
- Adds it to bookmarks.json

So add a guard clause in the toggle-bookmark handler to skip internal pages